fibreda/readme.md

```diff
@@ -0,0 +1,512 @@
+
+### 8.2 Badger layout
+
+Keys:
+
+* `d/<commitment>` → **blob** (big value; no TTL)
+* `p/<commitment>` → pointer: `"u:<YYYYMMDDHHmm>"` or `"c:<YYYYMMDDHHmm>"`
+* `bu/<YYYYMMDDHHmm>/<commitment>` → index for unconfirmed bucket
+* `bc/<YYYYMMDDHHmm>/<commitment>` → index for confirmed bucket
+
+Why this shape:
+
+- **Promotion is tiny**: just move the pointer and swap a small bucket entry; the 300 KB `d/*` value is **not** rewritten.
+- **GC is cheap**: at each minute boundary, iterate `bu/<nowmin>/*` and `bc/<nowmin>/*` and delete those items + their `d/*` + `p/*` keys.
+
+**Reasons**: promotion updates only tiny keys; GC is a minute-bucket sweep; no big-value rewrite on promotion.
+
+**Store API**
+
+```go
+type Store interface {
+  Put(ctx context.Context, key [32]byte, value []byte, safety time.Duration) error
+  Promote(ctx context.Context, key [32]byte) (promoted bool, err error)
+  Get(ctx context.Context, key [32]byte) ([]byte, bool, error)
+  RunGC(ctx context.Context)
+  Close() error
+}
+```
+
+**GC**: every minute boundary, sweep `bu/<now>/*` and `bc/<now>/*`, deleting bucket entries, pointer, and data; run Badger value-log GC opportunistically.
+
+### 8.3 Upload verification pipeline (mandatory)
+
+1. **Dedup** by `commitment` (if already present → `deduplicated=true`, return extant attestation if desired).
+2. Build **rsema1d verification context** from `rlc_orig`.
+3. For each row:
+
+    * Verify **Merkle proof** vs `commitment`.
+    * Verify **encoding soundness** using Context‑Based Verification with the RLC context.
+    * **Why RLC check**: detects adversarial encodings that pass Merkle but are off‑codeword.
+4. Persist rows with `expire_at = now + unconfirmed_ttl + safety_buffer` (batch write + sync).
+5. Compute `service_epoch_minute`; **sign** domain‑tagged bytes; return **attestation(s)**.
+
+### 8.4 Get pathway
+
+* Accept `bitmap` or `indices`. If neither provided, return **all** stored rows for the commitment.
+* Partial fulfillment allowed; include `missing_indices`.
+
+### 8.5 Promotion Listener
+
+* Subscribe to on-chain **PFB** events.
+* On observed `commitment`, call `Promote()` to move from `bu/*` to `bc/*` and update pointer `p/*`.
+* Optionally **issue a new attestation** for the confirmed period and expose via `GetAttestation(Commitment)` or piggyback on a promotion ack.
+
+### 8.6 API limits & fairness
+
+* Enforce `server_rows_per_message_limit` and maximum serialized request size.
+* **Rate-limit** ingress to **\~10 MiB/s** (configurable). With \~**300 KiB**/req worst-case, this is **\~33 RPS**. This is **conservative** because not all requests are max size; start here and raise with telemetry.
+* Include `backoff_ms` in responses to guide clients under pressure.
+
+### 8.7 Storage roadmap
+
+* **SQL deletes are not relevant in v1**. We **start with Badger**. Later we may prototype **SQL-light** and **epoch-bucket** stores and pick based on measured throughput and GC performance.
```

Comment:
I'm a bit confused as to why this is in the spec as there are not other roadmap distinctions